lib/iris/_concatenate.py

```diff
@@ -331,18 +359,20 @@ def __init__(self, cube):
         self.defn = cube.metadata
         self.data_type = cube.dtype
 
-        #
-        # Collate the dimension coordinate metadata.
-        #
+        # Populate dim- and aux-coord metadata.
+        self._collate_dim_metadata(cube)
```

Comment:
Another finnicky style point ...
I can see the point of separating the complicated bits into separate functions, but the effect of these 2 lines is now opaque.
Could we maybe return the data and assign it to self here to make it clearer ??  
I *think* those lists appended to aren't actually modified anywhere else, so we can make all this much clearer.
E.G.
```
    self.dim_metadata, self.dim_mapping = self._collate_dim_metadata(cube)
    self.aux_metadata, self.aux_coords_and_dims = self._collate_aux_metadata(cube)
```
(and don't pre-init them to empty lists above, either).